### ISSUE TITLE
Para Owner cannot unlock and deregister a Parachain when downgraded to Parathread

### ISSUE THREAD

**Issue by chrisdcosta** [paritytech/polkadot-sdk#4565]
### Is there an existing issue?

- [X] I have searched the existing issues

### Experiencing problems? Have you tried our Stack Exchange first?

- [X] This is not a support question.

### Description of bug

After a parachain has been downgraded to a parathread, the para owner (manager) should be able to deregister the parathread if so desired. The code hints that this should be possible however in practice there is a logic trap:

* A lock is applied to the parachain when it starts to produce blocks that is not lifted when the chain is downgraded.
* A parathread cannot unlock itself because it no longer produces blocks.
* The para owner is not allowed to remove the lock.
* A lock can therefore only be removed by Root.
* There is no appropriate governance track to unlock a parathread.

This appears to be a bug following [this question](https://substrate.stackexchange.com/questions/11485/what-is-the-opengov-track-required-to-deregister-a-parathread/11488#11488) on stack exchange.

The affected file is found at [polkadot-sdk/polkadot/runtime/common/src/paras_registrar/mod.rs](https://github.com/paritytech/polkadot-sdk/blob/master/polkadot/runtime/common/src/paras_registrar/mod.rs)

The context is that the para owner should not be able to remove a lock from a running parachain, because that is not decentralised. I agree with this.

However I would also argue that when a parachain is down-graded it may need to be unlocked to either perform a parachain swap or to de-register - either of those case involve some centralised decision making and therefore the para owner / manager must be given the power to unlock the parathread.

### Steps to reproduce

This can be performed on any non-production relaychain:

1. Register a parachain ID with an owner.

2. Add the PVF.

3. Upgrade to parachain for 1 lease period.

4. When the parachain automatically downgrades to a parathread following the end of the lease, attempt to de-register.

The Parathread cannot be de-registered due to a lock

5. Attempt to remove the lock by the para owner

The Para owner will get a BadOrigin error.

**Comment by xlc** [comment]
It will be a centralization risk if parathread/on-demand parachain owner can just remove it without any restrictions. I will say this is inteded behaviour and there isn't much we can do to improve without introducing new risks.

**Comment by bkchr** [comment]
@chrisdcosta you beat me to create an issue for this ;) It was on my list!

Generally, we should not just allow the parachain owner to deregister them as they like, as @xlc said. However, currently there is no way at all to unregister your chain if you are downgraded to a parathread, because we don't support this right now. In the future when we have on-demand, this will now be such a big problem, as the chain will always be able to buy one more block and deregister itself. In the future there will also not be anymore any distinguishing between on-demand or parachain on this level. All of them are parachains, just having a different way of acquiring blockspace.

**Comment by chrisdcosta** [comment]
@xlc I agree with your comment in the current implementation, because it provides no further checks and no distinction is made between the circumstances. This could be improved in order to provide more specific conditions.

There is another issue however - I think a PVF swap cannot be made either once a parachain has been downgraded, and this does require some form of central coordination between parties wishing to perform this - or indeed to address @bkchr point as new functionality becomes available, downgraded chains would also be unable to upgrade their own PVF to implement any necessary changes.

Lastly, it could be viewed that funds are unreasonably trapped by this situation and as there is no governance process to address it I think the best way in the interim is to provide code with specific conditions and a robust test. 

I have created a pull request #4568 for review. It contains some minor improvement noted in text.

**Comment by clangenb** [comment]
As we now have on demand, I think this issue is technically solved. The question is whether this is properly documented somewhere.
